### ISSUE TITLE
Move Terms and Conditions for Mapbox's locator overlay

### ISSUE THREAD

**Issue by eneerhut** [facebook/Rapid#1617]
### Description

We currently show and Terms and Conditions link the bottom right hand corner that [links to Mapbox](https://www.mapbox.com/about/maps). It's not clear to the user what the link relates to and it doesn't even link to terms currently. 

Some options to consider:
1. Display terms and conditions behind an 'I' if we are allowed to (proposed by @bhousel).
2. Utilize Meta's Canterbury style so that we can show labels at low zoom levels. Could ask Jonah for some advice here and it would mean we can swap out Mapbox's locator overlay.

**Comment by bhousel** [comment]
Good idea - I've never been really happy with how we show attributions for overlay layers.

If the user adds more overlays, we add more attributions down in this section, and it's not clear which overlay goes with which attribution text.  I also put the MapWithAI attribution text here too.

<img width="352" alt="Screenshot 2024-11-19 at 2 12 39 PM" src="https://github.com/user-attachments/assets/c54c7470-6900-44c5-9295-12c87763fead"><img width="417" alt="Screenshot 2024-11-19 at 2 12 48 PM" src="https://github.com/user-attachments/assets/966d3a5f-5e25-48ff-8214-30f1b414334e">


I think we could probably move them to a separate popup screen behind an ℹ️  button.

One other point about the Mapbox overlay - I'm fine with swapping it out for something else, but we currently only support raster for this, not vector, and it includes some roads and places and other details too. Not sure whether that would be a dealbreaker, (cc @jonahadkins)
